core/src/components/checkbox/checkbox.scss

```diff
@@ -87,7 +87,10 @@
   overflow: hidden;
 }
 
-:host(.in-item) .label-text-wrapper {
+// Checkboxes that are not slotted inside an item and are not used with a
+// stacked label should have margins equal to those of the label.
+:host(.in-item) .label-text-wrapper,
+:host(.in-item:not(.checkbox-label-placement-stacked):not([slot])) .native-wrapper {
```

Comment:
Slotted checkboxes are excluded because the following would otherwise result in increased item height:

```html
<ion-item>
  <ion-checkbox slot="start">Start</ion-checkbox>
  Checkbox
</ion-item>
```